TopMovies/Service/MovieService.swift

```diff
@@ -22,23 +22,38 @@ class MovieService: StoreSubscriber {
     switch state.movieCategoriesState.categoriesList {
     case .requested:
       mainStore.dispatch(MovieCategoriesAction.downloading)
-      movieAPI.topMovies { (result) in
+      movieAPI.allMovieCategories { result in
         switch result {
-        case let .success(categoriesList) where !categoriesList.isEmpty:
+        case let .success(categoriesList):
+          categoriesList
+            .map { MoviesDownloadingAction
+              .completed(MovieCategory(dto: $0).id,
+                         $0.results.map { Movie(dto: $0) })}
+            .forEach(mainStore.dispatch)
           mainStore.dispatch(MovieCategoriesAction
                               .completed(categories: categoriesList.map { MovieCategory(dto: $0) }))
-          categoriesList.first.map { category in
-            MoviesDownloadingAction.completed(
-              MovieCategory.ID(value: String(category.id)),
-              category.results.map { Movie(dto: $0) }
-            )
-          }.map(mainStore.dispatch)
         case let .failure(error):
           mainStore.dispatch(MovieCategoriesAction.failed(error: error))
-        default: break
         }
       }
     default: break
     }
+    state.categoryRequestsState.categoryRequests.forEach { categoryID, requestState in
+      switch requestState {
+      case .requested:
+        mainStore.dispatch(MoviesDownloadingAction.downloading(category: categoryID))
+        movieAPI.category(MovieCategoryRequest.init(rawValue: categoryID.value)!, page: 1) // FIX ME
```

Comment:
Just in case you actually have something to fix in the future, let's still follow Swift code style:

```swift
// TODO: Fix force unwrap
```